src/analyzing/AST_to_IL.ml

```diff
@@ -1108,6 +1108,17 @@ and stmt_expr env ?e_gen st =
   | G.Return (t, eorig, _) ->
       mk_s (Return (t, expr_opt env eorig)) |> add_stmt env;
       expr_opt env None
+  | G.DefStmt (ent, G.VarDef { G.vinit = Some e; vtype = _typTODO }) ->
+      let e = expr env e in
+      let lv = lval_of_ent env ent in
+      let i = mk_i (Assign (lv, e)) (Related (G.S st)) in
+      mk_s (Instr i) |> add_stmt env;
+      let eorig =
```

Comment:
Here I think you should just use `Related`. I think we do this `match` for `if` because `if` is a statement in the IL... I should have documented it (I hope there is a test that fails if we break that).

See

```ocaml
  | G.DefStmt (ent, G.VarDef { G.vinit = Some e; vtype = _typTODO }) ->
      let ss, e' = expr_with_pre_stmts env e in
      let lv = lval_of_ent env ent in
      ss @ [ mk_s (Instr (mk_i (Assign (lv, e')) (Related (G.S st)))) ]
```

Reply:
Done.